main.go

```diff
@@ -478,7 +478,7 @@ func runTests(perfStatsForTest *perfTestUtils.PerfStats, mode int, testSuite *te
 	}
 
 	// Kill the peak memory thread to avoid race condition when saving metrics.
-	chanQuitPkMem <- true
+	close(chanQuitPkMem)
 
```

Comment:
Setting "chanQuitPkMem <- true" here works if the unending for-loop in the goroutine is still running. However, short-circuiting the goroutine with "return" on error condition caused "fatal: deadlock" on the channel. On the other hand, attempting to short-circuit the goroutine with "chanQuitPkMem <- true" caused the process to hang at that point (it appeared to be a channel deadlock, but I never saw the fatal crash). The "return and close" pattern above works in both cases, and is a preferred best practice in golang literature.